hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/record/SingleTransactionRecordBuilderImpl.java

```diff
@@ -243,7 +251,10 @@ public SingleTransactionRecord build() {
         final Timestamp consensusTimestamp = HapiUtils.asTimestamp(consensusNow);
         final Timestamp parentConsensusTimestamp =
                 parentConsensus != null ? HapiUtils.asTimestamp(parentConsensus) : null;
-
+        // sort the automatic associations to match the order of mono-service records
+        automaticTokenAssociations.sort(
+                Comparator.<TokenAssociation>comparingLong(a -> a.tokenId().tokenNum())
+                        .thenComparingLong(a -> a.accountIdOrThrow().accountNum()));
```

Comment:
Also, given this `sort()`, appears we need to change `TokenCreateSpecs.validateNewTokenAssociations()` to assert 
```
                        getTxnRecord(creationTxn)
                                .hasPriority(recordWith()
                                        .autoAssociated(accountTokenPairs(List.of(
                                                Pair.of(fractionalCollector, A_TOKEN),
                                                Pair.of(selfDenominatedFixedCollector, A_TOKEN),
                                                Pair.of(otherSelfDenominatedFixedCollector, A_TOKEN),
                                                Pair.of(treasury, A_TOKEN))))),
```
[here](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/token/TokenCreateSpecs.java#L210).

Reply:
Ah thank you 🙏 for checking this. Changed